server/src/shared/interfaces/response-objects/BoardResponseObject.ts

```diff
@@ -12,6 +12,8 @@ export interface BoardResponseObject {
     moderator: string[];
   };
   slides: string[] | Types.ObjectId[];
+  currentUserPermission: string,
+  size: number,
   slide?: SlideResponseObject | string | SuperSlideDocument;
   createdAt?: Date | string | boolean | undefined;
```

Comment:
Why do `createdAt` and `updatedAt` have so many types here? Why not just use `Date` or `string`?

Reply:
@marcepanowyy ?

Reply:
And same with slide...

Reply:
When we leave createdAt & updatedAt as a Date we get 

```
TS2322: Type string | boolean | undefined is not assignable to type Date | undefined
Type string is not assignable to type Date
BoardResponseObject. ts(16, 3): The expected type comes from property createdAt which is declared here on type
```

This is because in Schema these fields have SchemaTimestampsConfig interface:
  
```
  interface SchemaTimestampsConfig {
    createdAt?: boolean | string;
    updatedAt?: boolean | string;
```
 
You may notice that it might be undefined (?), string, boolean or just a Date...

Btw why asking?

Reply:
Can you ensure that the frontend will not receive 'true'/'false' from this?

Reply:
idk why boolean can be a type of createdAt, we are always getting the string/Date so dont worry :)

Reply:
Okay, and also could you clarify the type of the slide object? We would like to receive a concrete object, not SlideResponseObject | string | SuperSlideDocument. What exactly will be sent?

Reply:
It depends on the endpoint you are hitting

Reply:
i think we could narrow the type to be just `SlideResponseObject | string`

Reply:
@Adam0s007 its not the scope of PR, we could discuss the response values tomorrow

Reply:
As long as this change doesn't affect the backend response (i.e., there are no inconsistencies in mapping to the appropriate interfaces), you're right — this will be outside the scope of this PR